### PR TITLE
fix: Update intermediate CA and PKI role when common_name is changed.

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Fixes: #367 

When the common name is changed a new CSR will be generated to get a new intermediate CA that will replace the old one and the pki role will be updated too.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/charms/vault_k8s/v0/vault_client.py [hunk 1]
```diff
@@ -353,19 +348,15 @@ def sign_pki_certificate_signing_request(
...
-    def create_pki_charm_role(self, role: str, allowed_domains: str, mount: str) -> None:
+    def create_or_update_pki_charm_role(self, role: str, allowed_domains: str, mount: str) -> None:
         """Create a role for the PKI backend."""
         self._client.secrets.pki.create_or_update_role(
             name=role,
             mount_point=mount,
             extra_params={
                 "allowed_domains": allowed_domains,
                 "allow_subdomains": True,
+                "allow_bare_domains": True,
```
Comment (gruyaume) [inline lib/charms/vault_k8s/v0/vault_client.py:359]: I don't know about adding this. This is unrelated to the bug plus I haven't seen users requesting this. I think we may only want to improve the config description to make it more obvious that only subdomains should be used.

**Comment by saltiyazan** [comment]
Turning PR to draft until integration test issues are figured out.

File: lib/charms/vault_k8s/v0/vault_client.py (hunk 1 above)
Comment (saltiyazan) [reply lib/charms/vault_k8s/v0/vault_client.py]: Alright, I removed. We can come back to it if it gets requested.
Improved the description of the config too.

Comment (gruyaume) [reply lib/charms/vault_k8s/v0/vault_client.py]: We'd still need to remove this content from the PR description

Comment (saltiyazan) [reply lib/charms/vault_k8s/v0/vault_client.py]: Sure :)

**Review by gruyaume — APPROVED** [review]
